examples/int-range/int_range.bal

```diff
@@ -0,0 +1,21 @@
+import ballerina/io;
+
+public function main() {
+    forEachWithRange(0, 5, 2);
+    forEachWithRange(5, 0, -2);
+}
+
+function forEachWithRange(rangeStart, rangeEnd, step) {
+    int[] loop = [];
+    index = 0;
+    // The integers returned by the iterator belong to the set S,where S is `{ rangeStart + step*i such that i >= 0 }`.
```

Comment:
How will the user know what the iterator is?

IMO, we should have parts in this example.

1. range function in foreach with positive step
2. range function in foreach with negative step
3. range function as an iterable and maybe used with while

We only need to talk about the iterator/iterable in the third case if at all.

Reply:
Third one is not added? Basically assigning the result of calling `int:range` to a variable rather than using it in-line.